apps/src/tutorialExplorer/README.md

```diff
@@ -0,0 +1,119 @@
+
+When visiting the page in a non-en language, it first shows a set of tiles for "Activities in your language", that is tutorials that we know have explicit support for that language.  Below this set of tiles is a button which, when pressed, reveals the larger set of tutorials and filter options.
+
+## Responsive
+
+The UI renders a layout that varies with different browser window widths.  A simple new system was introduced [here](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/apps/src/tutorialExplorer/responsive.jsx), and is inspired by Bootstrap, which we were using in other parts of our site at the time.
+
+There are a few functions provided.  Here is how one of them can be used:
+```
+width: getResponsiveValue({lg: 33.3333333, sm: 50, xs: 100})
+```
+
+Depending upon the current browser width, and whether that's considered "large", "small", or "extra-small", a different width can be applied to a given element.  (And in the case of the browser being at "medium", the function falls down to use the "small" value, since a "medium" one isn't provided.)
+
+## Algorithm
+
+At the core of the app is a filtering algorithm, implemented in the [`filterTutorials` function](
+https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/apps/src/tutorialExplorer/tutorialExplorer.js#L357-L477).  The comments describe its behavior with some detail.  Its role is to translate the user's choices in the user interface to an appropriate subset of the entire tutorial set.
+
+Care was taken to [unit test](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/tutorialExplorer/TutorialExplorerTest.js) this functionality.
+
+## Spreadsheet
+
+The page's [source HTML](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/pegasus/sites.v3/code.org/public/learn/index.haml#L120-L133) [initializes](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/pegasus/sites.v3/code.org/public/learn/index.haml#L140-L149) the **Tutorial Explorer** in JavaScript with a parameter containing all of the tutorial metadata.  That page is rendered by [Pegasus](https://github.com/code-dot-org/code-dot-org/tree/929fb8fffe8c70817630627df5766bf2c706d9fc/pegasus), and the tutorial metadata comes from an in-house gsheet which is used to generate [cdo-tutorials.csv](https://github.com/code-dot-org/code-dot-org/blob/929fb8fffe8c70817630627df5766bf2c706d9fc/pegasus/data/cdo-tutorials.csv) which is ingested daily into our source tree and used to seed the Pegasus database in each environment, alongside a variety of other gsheets' data.
+
+Management of the source gsheet is a significant task in itself, and the sheet has some internal complexity so that it can generate the appropriate values for the variety of `displayweight_*` and `popularityrank_*` columns (which correspond to "Recommended" vs. "Most popular" in the UI).
```

Comment:
Is there anything common and specific that will break this gsheet sync? Like issues we've had in the past but now someone (Eric maybe?) knows to sidestep?

Reply:
Great question.  It feels like a lot of possibilities, but no obvious, repeated ones come to mind.  And maybe that information can go into an internal doc alongside the gsheet itself.  But I'll ponder further.